Training/python/DataLoader.py

```diff
@@ -1,228 +1,259 @@
-            raise RuntimeError("chunk_size should be greater or equal to batch_size, and be proportional to it")
-
-
-        self.net_config = net_config
-        self.batch_size = batch_size
-        self.chunk_size = chunk_size
-        self.n_passes = n_passes
-        self.max_queue_size = max_queue_size
-        self.return_grid = return_grid
-        self.has_validation_set = validation_size is not None
-
-        all_files = [ f.replace('\\', '/') for f in sorted(glob.glob(file_name_pattern)) ]
-        evt_left = max_data_size
-        val_evt_left = validation_size
-        self.file_entries = []
-        self.data_size = 0
-        self.validation_size = 0
-        self.steps_per_epoch = 0
-        self.validation_steps = 0
-        for file_name in all_files:
-            file_size = DataLoader.GetNumberOfEntries(file_name, 'taus')
-            file_entry = FileEntry(file_name, file_size, batch_size, evt_left, val_evt_left)
-            if evt_left is not None:
-                evt_left -= file_entry.size
-            if val_evt_left is not None:
-                val_evt_left -= file_entry.val_size
-            self.data_size += file_entry.size
-            self.validation_size += file_entry.val_size
-            self.steps_per_epoch += file_entry.steps
-            self.validation_steps += file_entry.val_steps
-            self.file_entries.append(file_entry)
-        self.total_size = self.data_size + self.validation_size
-
-        if val_evt_left is not None and val_evt_left != 0:
-            raise RuntimeError("Insufficient number of events to create validation set.")
-        if self.data_size == 0:
-            raise RuntimeError("Insufficent number of events to create data set.")
-
-    def generator(self, primary_set = True, return_truth = True, return_weights = True):
-        file_entries = []
-        if primary_set:
-            for entry in self.file_entries:
-                if entry.size > 0:
-                    file_entries.append( (entry.file_name, entry.tau_begin, entry.tau_end) )
-            steps_per_epoch = self.steps_per_epoch
-        else:
-            for entry in self.file_entries:
-                if entry.val_size > 0:
-                    file_entries.append( (entry.file_name, entry.val_tau_begin, entry.val_tau_end) )
-            steps_per_epoch = self.validation_steps
-
-        queue = Queue(maxsize=self.max_queue_size)
-        current_pass = 0
-        while self.n_passes < 0 or current_pass < self.n_passes:
-            thread = Thread(target=LoaderThread,
-                     args=( file_entries, queue, self.net_config, self.batch_size, self.chunk_size,
-                            return_truth, return_weights, self.return_grid ))
-            thread.daemon = True
-            thread.start()
-            for step_id in range(steps_per_epoch):
-                item = queue.get()
-                yield item
-            thread.join()
+    def ListToVector(l, elem_type):
```

Comment:
`ListToVector` is a general purpose function, move its definition outside of DataLoader